nilearn/masking.py

```diff
@@ -607,32 +613,56 @@ def compute_brain_mask(target_img, threshold=.5, connected=True,
         Controls the amount of verbosity: higher numbers give
         more messages
 
+    mask_type : {'whole-brain', 'gm', 'wm'}, optional
+        Type of mask to be computed. Default = 'whole-brain'
+
+    **kwarg: optional argument
+        resolution (int) is taken here as an optional argument to resample the
+        original template at the specified resolution
+
     Returns
     -------
     mask : nibabel.Nifti1Image
         The whole-brain mask (3D image)
     """
     if verbose > 0:
-        print("Template mask computation")
+        msg_to_print = "Template " + mask_type + " mask computation"
+        print(msg_to_print)
 
     target_img = _utils.check_niimg(target_img)
 
-    from .datasets import load_mni152_brain_mask
-    template = load_mni152_brain_mask()
+    if mask_type == 'whole-brain':
+        template = load_mni152_template()
+    elif mask_type == 'gm':
+        template = load_mni152_gm_template()
+    elif mask_type == 'wm':
+        template = load_mni152_wm_template()
+    else:
+        raise ValueError("Unknown mask type {}.".format(mask_type))
+
+    # Change the resolution of the template
+    if kwarg:
+        if 'resolution' in kwarg.keys():
+            target_resolution = kwarg['resolution']
+        else:
+            raise KeyError('missing resolution argument.')
```

Comment:
This will raise an error if someone passes another kwarg without changing resolution (which could happen if we support additional kwargs).
I would prefer something like:

```python
target_resolution = kwargs.pop('resolution', 1)

# If no target resolution is provided, it will default to 1
# and resample_img will exit early
template = cache(resampling.resample_img, memory)(
        template, np.eye(3) * target_resolution)
```

Reply:
as discussed above `resolution` will be a formal parameter so there will be no need to look for it in a dict